hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -121,36 +133,21 @@ class HandlerPaths(using Elaborator.State):
   val handleBlockImplPath: Path = runtimePath.selSN("handleBlockImpl")
   val stackDelayClsPath: Path = runtimePath.selSN("StackDelay")
   val topLevelEffectPath: Path = runtimePath.selSN("topLevelEffect")
-  
-  def isHandlerClsPath(p: Path) =
-    (p eq contClsPath)  || (p eq stackDelayClsPath) || (p eq effectSigPath)
+  val ctorEffectPath: Path = runtimePath.selSN("ctorEffect")
+  val enterHandleBlockPath: Path = runtimePath.selSN("enterHandleBlock")
+  val stackDepthIdent = new Tree.Ident("stackDepth")
+  val stackDepthPath: Path = runtimePath.selN(stackDepthIdent)
+  val fnLocalsPath: Path = runtimePath.selSN("FnLocalsInfo").selSN("class")
+  val localVarInfoPath: Path = runtimePath.selSN("LocalVarInfo").selSN("class")
+  val unwindPath: Path = runtimePath.selSN("unwind")
+  val resumePc: Path = runtimePath.selSN("resumePc")
+  val resumeIdx: Path = runtimePath.selSN("resumeIdx")
+  val resumeValueIdent = new Tree.Ident("resumeValue")
+  val resumeValue: Path = runtimePath.selN(resumeValueIdent)
```

Comment:
I think the name `resumeValue` is a bit confusing, as that suggests it's the value you're passing to the resumption function. I think a better name would be `curEffect` or `curEffectSig`.

Reply:
When the function calls unwind, the resumeValue is not used by the runtime to determine the effect, rather it's passed explicitly to the unwind function

Reply:
OK. I see